drivers/sensor/tdk/icp101xx/icp101xx.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2023 TDK Invensense
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_ICP101XXX_ICP101XXX_H_
+#define ZEPHYR_DRIVERS_SENSOR_ICP101XXX_ICP101XXX_H_
+
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include "Devices\Drivers\ICP101xx\Icp101xx.h"
+#include "Devices\Drivers\ICP101xx\Icp101xxSerif.h"
+
+#define DT_DRV_COMPAT invensense_icp101xx
+
+#define ICP101XX_BUS_I2C DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+
+typedef struct {
+	int raw_pressure;
+	int raw_temperature;
+	float pressure;
```

Comment:
floats require some special care in Zephyr as floats are *not* always stashed on context swaps, not always supported by the hardware (cortex-m0), and generally not accepted in sensor drivers for those reasons

Reply:
Do you confirm that using the k_float_enable/k_float_disable within the driver would make things safer?
As said in other comments, the official TDK driver for ICP101xx sensor relies on float.